### ISSUE TITLE
hide_urlbar_first_row.css isn't working

### ISSUE THREAD

**Issue by april** [MrOtherGuy/firefox-csshacks#83]
I'm not sure exactly what it is for `visit` links, but this works to hide _Search With..._ as an option:

```
#urlbar div[type="tabtosearch"] {
  display: none;
}
```

**Comment by MrOtherGuy** [comment]
As far as I can tell, the style is working fine. On what circumstances it isn't?

Note that the first row is highlighted as selected even if it is not shown. Thus, if you hit down-button once and it then highlights the first *visible* row then you can conclude that it has indeed hidden then the first row.

**Comment by MrOtherGuy** [comment]
What I meant by that is that you might be experiencing a scenario where the dropdown has multiple "search with" -items but the style only hides the first one. That is intended behavior.

On the other hand, if you mean the behavior related to the new tab-to-search options then I don't really know what sort of behavior one would expect. When you invoke tab-to-search, Firefox sets the tab-to-search result as selected, but it doesn't modify the ordering of the results list. So the tab-to-search element will probably be the second result. Thus, it is indeed not hidden in that scenario, but that's because the selected row is not the first row - not because the style isn't working. It is infact working, since it is hiding the first row which is likely a "visit <page>" item which you can activate using "up" arrowkey.

There are couple of different behaviors we could use to handle tab-to-search scenarios. I have tried several and I think the most sane option would be essentially what you proposed - `.urlbarView-row[type="tabtosearch"]{ display: none; }`

The only downside I can think of is that adding that would also remove that row *before* the tab (or down-arrow) has been pressed which would activate the tab-to-search "mode". Thus the user is not informed that this result exists. However, I don't think that's a very big deal because if the user is using this custom stylesheet in the first place then they probably know the behavior Firefox has.

tl;dr - I propose that the style should hide the first result just as it currently does, but also hide all [tabtosearch] rows. How does that sound?

**Comment by april** [comment]
Ahhhh, you're right. This is how it looks for me without any CSS in Nightly 84.0a1:

<img width="506" alt="Screen Shot 2020-10-25 at 7 08 38 AM" src="https://user-images.githubusercontent.com/230528/97106665-15fe7a00-1691-11eb-8af0-be2eba3ace07.png">

And here it is with your CSS:

<img width="550" alt="Screen Shot 2020-10-25 at 7 07 23 AM" src="https://user-images.githubusercontent.com/230528/97106670-1d258800-1691-11eb-8f82-0897c4097293.png">

As well as my CSS:

<img width="630" alt="Screen Shot 2020-10-25 at 7 05 54 AM" src="https://user-images.githubusercontent.com/230528/97106730-5100ad80-1691-11eb-928a-bfa77ea7d839.png">

The two of them together though do look perfect:

<img width="549" alt="Screen Shot 2020-10-25 at 7 11 39 AM" src="https://user-images.githubusercontent.com/230528/97106744-6544aa80-1691-11eb-9cd0-ec9adda72bc8.png">

So yeah, that means this:

```css
#urlbar[usertyping] .urlbarView-row:not([type]):first-child,
#urlbar[usertyping] .urlbarView-row[type="search"]:first-child {
  display: none;
}

#urlbar div[type="tabtosearch"] {
  display: none;
}
```

Removes any suggestions and instead gets right into your history.

**Comment by MrOtherGuy** [comment]
Alright, thanks for this suggestion.
